tests/main/snap-session-agent-unavailable-to-snaps/task.yaml

```diff
@@ -0,0 +1,55 @@
+summary: Check that snaps can not access the session agent
+
+systems:
+    # Ubuntu 14.04 does not have a complete systemd implementation
+    - -ubuntu-14.04-*
+    # Systemd on CentOS 7/Amazon Linux 2 does not have the user@uid unit
+    - -amazon-linux-2-*
+    - -centos-7-*
+
+environment:
+    TEST_UID: $(id -u test)
+    USER_RUNTIME_DIR: /run/user/${TEST_UID}
+
+prepare: |
+    # Ensure that snapd.session-agent.socket is enabled.  This may not
```

Comment:
The test only executes on systems where `snap debug confinement` returns `strict`. You could also short circuit prepare/restore steps on other systems too.